src/routes/Settings/SettingsIndexView.js

```diff
@@ -1,7 +1,164 @@
 // @flow
 
-function SettingsIndexView() {
-  return null
+import React, { PureComponent } from 'react'
+
+import { JText } from 'components/base'
+import { SettingsGrid, SettingsGridCard } from 'components'
+
+import { divideThousands } from 'utils/numbers'
+import { formatBoolean, formatCurrency, formatLanguage } from './utils'
+
+type Props = {|
+  ...SettingsState,
+  networkName: ?string,
+  wallet: ?Wallet,
+|}
+
+class SettingsIndexView extends PureComponent<Props, *> {
+  render() {
+    const {
+      localCurrencyCode,
+      defaultGasPrice,
+      systemLanguageCode,
+      hasPinCode,
+      networkName,
+      wallet,
+    } = this.props
+
+    if (!wallet) {
+      return null
+    }
+
+    const { name: walletName, customType, isReadOnly, mnemonicOptions } = wallet
+    const isFullMnemonic = customType === 'mnemonic' && !isReadOnly
+    const derivationPath = mnemonicOptions ? mnemonicOptions.derivationPath : null
+    const passphrase = mnemonicOptions ? mnemonicOptions.passphrase : null
+    const isNetworkName = Boolean(networkName)
+    const isWalletName = Boolean(walletName)
+
```

Comment:
isWalletName is always true

Reply:
But, it has `?string` type. Are you sure?

Reply:
btw, do not forget that `Boolean('') === false`